src/transformers/models/llama/modeling_llama.py

```diff
@@ -807,22 +815,43 @@ def _init_weights(self, module):
             if module.padding_idx is not None:
                 module.weight.data[module.padding_idx].zero_()
 
-    def _setup_cache(self, cache_cls, max_batch_size, max_cache_len: Optional[int] = None):
-        if self.config._attn_implementation == "flash_attention_2" and cache_cls == StaticCache:
+    def _setup_cache(self, cache_cls, max_batch_size, generation_config):
+        if self.config._attn_implementation == "flash_attention_2" and (
+            cache_cls == StaticCache,
```

Comment:
The `(cache_cls == StaticCache, cache_cls == PagedAttentionCache)` will be always considered as positive in the condition statement because it is a `Tuple`.